metagraph/src/annotation/representation/annotation_matrix/annotation_matrix.hpp

```diff
@@ -39,6 +42,8 @@ class StaticBinRelAnnotator : public MultiLabelEncoded<Label> {
 
     const BinaryMatrixType& get_matrix() const override { return *matrix_; };
 
+    BinaryMatrixType&& release_matrix()  { return std::move(*matrix_); };
```

Comment:
Not sure I understand what you mean.

With
```c++
BinaryMatrixType* release_matrix()  { return matrix_.release(); };
```
you would do `std::unique_ptr<BinaryMatrixType> new_matrix(annotator.release_matrix());`

And you don't need to free anything.

How is this less readable?

Reply:
Maybe I should say "more verbose" rather than "less readable". Here is what the caller currently looks like:
```  
std::vector<std::unique_ptr<bit_vector>> columns 
    = annotator.release_matrix().diffs().release_columns();
```

If I made this change, the caller would have to construct a unique_ptr:

```  
std::vector<std::unique_ptr<bit_vector>> columns
            = std::unique_ptr<binmat::RowDiff<binmat::ColumnMajor>>(annotator.release_matrix())
                      .diffs()
                      .release_columns();
```

Reply:
Hm, I see. Then maybe return `std::unique_ptr<BinaryMatrixType>`, so you can still write
```c++
std::vector<std::unique_ptr<bit_vector>> columns 
    = annotator.release_matrix()->diffs().release_columns();
```

Reply:
It requires move constructors to be implemented.

For you example, you don't have to call the second release(), you just need to dereference and move
```c++
return std::make_unique<RowDiffBRWTAnnotator>(
            std::make_unique<RowDiff<BRWT>>(nullptr, std::move(*annotator->release_matrix()), ""),
            annotator->get_label_encoder());
```

Reply:
also, if you need a pointer, this
```
std::make_unique<BRWT>(std::move(annotator->release_matrix()));
```
becomes simply this
```
annotator->release_matrix();
```

Reply:
It is generally easier to get a reference from a pointer than the other way around. So if we already have a pointer, it's good to keep it as a pointer without reinitializing the object with move constructors.
Then, if a reference is needed, the user will be responsible to convert it.